apps/dcos_overlay/src/dcos_overlay_poller.erl

```diff
@@ -120,51 +129,108 @@ parse_response(State0 = #state{known_overlays = KnownOverlays}, AgentInfo) ->
     State1 = State0#state{ip = IP1},
     Overlays = maps:get(<<"overlays">>, AgentInfo, []),
     NewOverlays = Overlays -- KnownOverlays,
-    lists:foldl(fun add_overlay/2, State1, NewOverlays).
+    lists:foldl(
+      fun(Overlay, {ok, StateAcc}) ->
+          case add_overlay(Overlay, StateAcc) of
+              {ok, _} = Result ->
+                  prometheus_counter:inc(
+                    overlay, networks_added_total, [], 1),
+                  Result;
+              {error, ErrorMsg} = Error ->
```

Comment:
May I kindly ask you to explicitly match and return tuples _everywhere_? It'd be much easier to follow snd understand the logic and returning types in that case. i.e.

```
   case foobar() of
       {ok, X} ->
           {ok, X};
       {error, Error} ->
           {error, Error}
    end.
```